company/models.py

```diff
@@ -22,6 +22,208 @@ def formfield(self, **kwargs):
+    plRegistered_notes = models.TextField(_("Więcej nt. miejsca rejestracji"), null=True, blank=True)
+    plNotGlobEnt_notes = models.TextField(_("Więcej nt. struktury kapitałowej"), null=True, blank=True)
+
+    nip = models.CharField(max_length=10, db_index=True, null=True, blank=True, verbose_name=_("NIP/Tax ID"))
+    address = models.TextField(null=True, blank=True, verbose_name=_("Adres"))
+
+    is_friend = models.BooleanField(
+        default=False, verbose_name=_("Przyjaciel Poli"), choices=((True, _("Tak")), (False, _("Nie")))
+    )
+    objects = CapitalGroupQuerySet.as_manager()
+
+    def to_dict(self):
+        dict = model_to_dict(self)
+        return dict
+
+    def get_absolute_url(self):
+        return reverse('company:capital-group-detail', args=[self.pk])
+
+    def locked_by(self):
+        return concurency.locked_by(self)
+
+    def __str__(self):
+        return self.common_name or self.official_name or self.name
+
+    def get_sources(self, raise_exp=True):
+        ret = {}
+        if not self.sources:
+            return ret
+
+        lines = self.sources.splitlines()
+        for line in lines:
+            line = line.strip()
+            if line == '':
+                continue
+            s = line.split('|')
+            if s.__len__() != 2:
+                if raise_exp:
+                    raise ValidationError(
+                        'Pole >Źródła< powinno składać się '
+                        'linii zawierających tytuł odnośnika'
+                        ' i odnośnik odzielone znakiem | (pipe)'
+                    )
+                else:
+                    continue
+            if s[0] in ret:
+                if raise_exp:
+                    raise ValidationError(f'Tytuł odnośnika >{s[0]}< występuje więcej niż raz')
+                else:
+                    continue
+            ret[s[0]] = s[1]
+
+        return ret
+
+    def clean(self, *args, **kwargs):
+        if self.verified:
+            YOU_CANT_SET_VERIFIED = (
+                'Nie możesz oznaczyć grupy kapitałowej jako ' 'zweryfikowana jeśli pole >{}< jest nieustalone'
+            )
+            if self.plCapital is None:
+                raise ValidationError(YOU_CANT_SET_VERIFIED.format('udział kapitału polskiego'))
+            if self.plWorkers is None:
+                raise ValidationError(YOU_CANT_SET_VERIFIED.format('miejsce produkcji'))
+            if self.plRnD is None:
+                raise ValidationError(YOU_CANT_SET_VERIFIED.format('wysokopłatne miejsca pracy'))
+            if self.plRegistered is None:
+                raise ValidationError(YOU_CANT_SET_VERIFIED.format('miejsce rejestracji'))
+            if self.plNotGlobEnt is None:
+                raise ValidationError(YOU_CANT_SET_VERIFIED.format('struktura kapitałowa'))
+            self.get_sources()
+
+        super().clean(*args, **kwargs)
+
+    def as_company(self, pk):
+        return Company(
+            pk=pk,
```

Comment:
Zamiast nadpisywać pk to lepiej stworzyć jakieś sztuczne np. `CG-{capitalGroup.pk}`, aby przypadkiem nie nadpisać pola